pylsp_ruff/ruff_lint.py

```diff
@@ -0,0 +1,265 @@
+import json
+import logging
+import sys
+from pathlib import PurePath
+from subprocess import PIPE, Popen, SubprocessError
+
+from pylsp import hookimpl, lsp
+from pylsp._utils import find_parents
+from pylsp.workspace import Document, Workspace
+
+# Use built-in tomllib for python>=3.11
+if sys.version_info >= (3, 11):
+    try:
+        import tomllib
+    except ImportError:
+        import tomli as tomllib
+else:
+    import tomli as tomllib
+
+log = logging.getLogger(__name__)
+
+UNNECESSITY_CODES = {
+    "F401",  # `module` imported but unused
+    "F504",  # % format unused named arguments
+    "F522",  # .format(...) unused named arguments
+    "F523",  # .format(...) unused positional arguments
+    "F841",  # local variable `name` is assigned to but never used
+}
+
+
+@hookimpl
+def pylsp_settings():
+    # flake8 and pycodestyle disabled by default with this plugin
+    return {
+        "plugins": {
+            "ruff": {
+                "enabled": True,
+                "config": None,
+                "exclude": None,
+                "executable": "ruff",
+                "ignore": None,
+                "lineLength": None,
+                "perFileIgnores": None,
+                "select": None,
+            },
+            "flake8": {"enabled": False},
+            "pycodestyle": {"enabled": False},
+        }
+    }
+
+
+@hookimpl
+def pylsp_lint(workspace: Workspace, document: Document) -> list:
```

Comment:
I’m happy to help extend this to support Quick Fix actions for autofixable errors (which I assume are supported by pylsp)!